server/src/smart-scheduling-routes.ts

```diff
@@ -176,10 +228,42 @@ export async function listSchedules(
   res.header("Content-Type", "application/fhir+ndjson").send(
     providers
       .map((provider) => {
+        const extension: any[] = [
+          {
+            url: EXTENSION_HAS_AVAILABILITY,
+            valueCode: formatHasAvailability(provider.availability),
+          },
+        ];
+
+        // TODO: produce separate schedules for each combination of products
+        // and dose that appear in `availability.slots/capacity`.
+        if (provider.availability?.products) {
+          for (const product of provider.availability.products) {
+            extension.push({
+              url: EXTENSION_VACCINE_PRODUCT,
+              valueCoding: {
+                system: SYSTEM_CVX,
+                code: CVX_CODES[product],
+                display: PRODUCT_NAMES[product],
+              },
+            });
+          }
+        }
+        if (provider.availability?.doses) {
+          const doseNumbers = provider.availability.doses.flatMap(
+            (dose: string) => DOSE_NUMBERS[dose] || []
+          );
+          for (const doseNumber of new Set(doseNumbers)) {
```

Comment:
nit: It would be nice (for some definition of "nice") to ensure that the dose numbers are sorted here. Not sure it's worth messing with given the significance of the rest of the diff, but if you do, it might be simpler just to loop through `DOSE_NUMBERS.all_doses` (which will presumably stay ordered) and check if each of the values is in this set.

Reply:
Can do that. I think this would also work, though it’s not very fun to read:

```suggestion
          for (const doseNumber of [...new Set(doseNumbers)].sort()) {
```